app/src/main/java/com/banuba/quickstart_c_api/OffscreenEffectPlayerImage.java

```diff
@@ -1,45 +1,125 @@
+
+    OffscreenEffectPlayerImage(ImageProxy imageProxy) {
+        mWidth = imageProxy.getWidth();
+        mHeight = imageProxy.getHeight();
+        mPixelFormat = imageProxy.getImage().getFormat();
+
+        ImageProxy.PlaneProxy planeProxy = imageProxy.getPlanes()[0];
+        mRowStride0 = planeProxy.getRowStride();
+        mPlane0 = planeProxy.getBuffer();
+        mPixelStride0 = planeProxy.getPixelStride();
+        int planesNumber = imageProxy.getPlanes().length;
+        if (planesNumber > 1) {
+            planeProxy = imageProxy.getPlanes()[1];
+            mRowStride1 = planeProxy.getRowStride();
+            mPlane1 = planeProxy.getBuffer();
+            mPixelStride1 = planeProxy.getPixelStride();
+            if (planesNumber > 2) {
+                planeProxy = imageProxy.getPlanes()[2];
+                mRowStride2 = planeProxy.getRowStride();
+                mPlane2 = planeProxy.getBuffer();
+                mPixelStride2 = planeProxy.getPixelStride();
+            } else {
+                mRowStride2 = 0;
+                mPlane2 = null;
+                mPixelStride2 = 0;
+            }
+        } else {
+            mRowStride1 = 0;
+            mPlane1 = null;
+            mPixelStride1 = 0;
+            mRowStride2 = 0;
+            mPlane2 = null;
+            mPixelStride2 = 0;
+        }
     }
 
-    public ImageInfo mImageInfo;
-    public ByteBuffer mImageZero = null;
-    public ByteBuffer mImageFirst = null;
-    public ByteBuffer mImageSecond = null;
-}
+    OffscreenEffectPlayerImage(byte[] rgbPlane, int width, int height) {
+        assert rgbPlane != null;
+        mPlane0 =  ByteBuffer.wrap(rgbPlane);
+        mPlane1 =  null;
+        mPlane2 =  null;
+        mWidth = width;
+        mHeight = height;
+
+        mRowStride0 = 0;
+        mRowStride1 = 0;
+        mRowStride2 = 0;
+        mPixelStride0 = 0;
+        mPixelStride1 = 0;
+        mPixelStride2 = 0;
+        mPixelFormat = 0;
+    }
+
+    OffscreenEffectPlayerImage(byte[] yPlane, byte[] uvPlane, int width, int height) {
+        assert yPlane != null && uvPlane != null;
+        mPlane0 =  ByteBuffer.wrap(yPlane);
+        mPlane1 =  ByteBuffer.wrap(uvPlane);
+        mPlane2 =  null;
+        mWidth = width;
+        mHeight = height;
+
+        mRowStride0 = 0;
+        mRowStride1 = 0;
+        mRowStride2 = 0;
+        mPixelStride0 = 0;
+        mPixelStride1 = 0;
+        mPixelStride2 = 0;
+        mPixelFormat = 0;
+    }
 
-class ImageInfo {
-    ImageInfo() {
-        width = 0;
-        height = 0;
-        inputOrientation = 0;
-        outputOrientation = 0;
-        requireMirroring = false;
-        rowStride0 = 0;
-        rowStride1 = 0;
-        rowStride2 = 0;
-        pixelStride0 = 0;
-        pixelStride1 = 0;
-        pixelStride2 = 0;
-        pixelFormat = 0;
-        imageFormat = 0;
+    OffscreenEffectPlayerImage(byte[] yPlane, byte[] uPlane, byte[] vPlane, int width, int height) {
```

Comment:
change to:
OffscreenEffectPlayerImage(final byte[] yPlane, final int yPlaneStride, final byte[] uPlane, final int uPlaneStride, final byte[] vPlane, final int vPlaneStride,  int width, int height)

Reply:
And this constructor will create an Image with i420 format